### PR TITLE
Enable creation of upgraded master during Kubernetes upgrade cluster operation

### PR DESCRIPTION
**Pull request by amanohar**
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/647)

### CODE REVIEW

File: parts/kubernetesparams.t [hunk 1]
```diff
@@ -133,4 +133,19 @@
         "description": "The Service Principal Client Secret."
       },
       "type": "securestring"
+    },
+    "masterOffset": {
+      "defaultValue": 0,
+      "allowedValues": [
+        1,
```
Comment (anhowe) [inline parts/kubernetesparams.t]: did you mean default of 1 or 0,1,2,3,4

File: pkg/acsengine/transform.go [hunk 2]
```diff
@@ -192,3 +195,63 @@ func removeImageReference(logger *logrus.Entry, resourceProperties map[string]in
...
+			if !ok {
+				logger.Warnf("Template improperly formatted")
+				continue
+			}
+
+			if strings.EqualFold(resourceType, vmResourceType) &&
+				strings.Contains(resourceName, "variables('masterVMNamePrefix')") {
+				resourceProperties, ok := resourceMap[propertiesFieldName].(map[string]interface{})
+				if !ok {
+					logger.Warnf("Template improperly formatted")
```
Comment (anhowe) [inline pkg/acsengine/transform.go]: What does it mean if the template if not properly formatted?  Does it halt?

File: parts/kubernetesparams.t (hunk 1 above)
Comment (amanohar) [inline parts/kubernetesparams.t]: Default should be 1. Will fix.

File: pkg/acsengine/transform.go (hunk 2 above)
Comment (amanohar) [inline pkg/acsengine/transform.go]: I was following the convention of other methods in this file of continuing. I haven't actively given it any thought (yet). I am going to leave it as is for now. This is one of those error cases that need handling.

**Comment by amanohar** [comment]
@acs-bot test this please

File: parts/kubernetesparams.t [hunk 3]
```diff
@@ -133,4 +133,19 @@
...
+      "defaultValue": 1,
+      "allowedValues": [
+        1,
+        2,
+        3,
+        4,
+        5
+      ],
+      "metadata": {
+        "description": "The offset into the master pool where to start creating master VMs.  This value can be from 0 to 4, but must be less than masterCount."
```
Comment (anhowe) [inline parts/kubernetesparams.t]: 1 to 5? and "less than masterCount"

Comment (amanohar) [inline parts/kubernetesparams.t]: Fixed.

**Comment by amanohar** [comment]
@acs-bot test this please

**Comment by amanohar** [comment]
@acs-bot test this please

**Pull request merged**